src/schemas/pubmatter.typ

```diff
@@ -0,0 +1,142 @@
+    institution: z.string(optional: true)
+  ),
+  pre-transform: coerce.dictionary((it)=>(name: it))
+)
+
+#let abstracts = z.dictionary(
+  (
+    title: z.string(default: "Abstract"),
+    content: z.content(),
+  ),
+  pre-transform: coerce.dictionary(it=>(content: it))
+)
+
+#let coerce-license(self, it) = {
+  if ( it in ("CC0", "CC0-1.0")){
+    return (
+      id: "CC0-1.0",
+      url: "https://creativecommons.org/licenses/zero/1.0/",
+      name: "Creative Commons Zero v1.0 Universal",
+    ) 
+  } else if ( it in ("CC-BY", "CC-BY-4.0")){
+    return (
+      id: "CC-BY-4.0",
+      url: "https://creativecommons.org/licenses/by/4.0/",
+      name: "Creative Commons Attribution 4.0 International",
+    )
+  } else if ( it in ("CC-BY-NC", "CC-BY-NC-4.0")){
+    return (
+      id: "CC-BY-NC-4.0",
+      url: "https://creativecommons.org/licenses/by-nc/4.0/",
+      name: "Creative Commons Attribution Non Commercial 4.0 International"
+    )
+  } else if ( it in ("CC-BY-NC-SA", "CC-BY-NC-SA-4.0")){
+    return (
+      id: "CC-BY-NC-SA-4.0",
+      url: "https://creativecommons.org/licenses/by-nc-sa/4.0/",
+      name: "Creative Commons Attribution Non Commercial Share Alike 4.0 International",
+    )
+  } else if ( it in ("CC-BY-ND", "CC-BY-ND-4.0")){
+    return (
+      id: "CC-BY-ND-4.0",
+      url: "https://creativecommons.org/licenses/by-nd/4.0/",
+      name: "Creative Commons Attribution No Derivatives 4.0 International",
+    )
+  } else if ( it in ("CC-BY-NC-ND", "CC-BY-NC-ND-4.0")){
+    return (
+      id: "CC-BY-NC-ND-4.0",
+      url: "https://creativecommons.org/licenses/by-nc-nd/4.0/",
+      name: "Creative Commons Attribution Non Commercial No Derivatives 4.0 International",
+    )
+  }
+  return it
+}
+
+#let pubmatter = z.dictionary(
+  aliases: (
+    "author": "authors",
+    "running-title": "short-title",
+    "running-head": "short-title",
+    "affiliation": "affiliations",
+    "abstract": "abstracts",
+    "date": "dates"
+  ),
+  (
+    title: z.content(optional: true),
+    subtitle: z.content(optional: true),
+    short-title: z.string(optional: true),
+    authors: z.array(author, pre-transform: coerce.array),
+    affiliations: z.array(affiliation, pre-transform: coerce.array),
+    abstracts: z.array(abstracts, pre-transform: coerce.array),
+    citation: z.content(optional: true),
+    open-access: z.boolean(optional: true),
+    venue: z.content(optional: true),
+    doi: z.string(optional: true),
+    keywords: z.array(z.string()),
+    license: z.dictionary(
+      optional: true,
+      (
+        id: z.string(),
+        url: z.string(),
+        name: z.string()
+      ),
+      pre-transform: coerce-license
+    ),
+    dates: z.array(
+      z.dictionary(
+        (
+          type: z.content(optional: true),
+          date: z.date(pre-transform: coerce.date)
+        ),
+        pre-transform: coerce.dictionary((it)=>(date: it))
+      ),
+      pre-transform: coerce.array
+    ),
+  )
+)
```

Comment:
This seems very specific to publication, which may not be useful to downstream users at all, perhaps this is better left as a supplemental package?

Reply:
Initially, this was going to be upstreamed into pubmatter's package, but the reasoning slowly morphed into the idea of providing well-thought out commonly-used schemas (such as authors, or abstracts) for other templates to implement to get a ball rolling on a form of opt-in standard for template arguments.

It's a good point that it could be in a separate package, but I'd worry about potential version conflicts in the far future. Keeping it in the same package ensures that community schemas are always compatible with the version of valkyrie they ship with.

Reply:
> but I'd worry about potential version conflicts in the far future

These will hardly be a problem now. Currently, packages are all pinned and in the far future if we do actually get to have a semver dependency resolver in Typst, it will likely have custom types by then.

I don't think it makes much sense to provide more than the building blocks and actually common types in this package itself.

Reply:
How about keeping author and abstracts, and upstreaming affiliations and pubmatter?

Reply:
I've elected to retain only the author schema.